litr/src/main/java/com/linkedin/android/litr/thumbnails/ThumbnailExtractParameters.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021 LinkedIn Corporation
+ * All Rights Reserved.
+ *
+ * Licensed under the BSD 2-Clause License (the "License").  See License in the project root for
+ * license information.
+ */
+package com.linkedin.android.litr.thumbnails
+
+import android.graphics.Point
+import android.net.Uri
+import com.linkedin.android.litr.ExperimentalThumbnailsApi
+import com.linkedin.android.litr.render.ThumbnailRenderer
+
+/**
+ * Request parameters for thumbnail extraction.
+ */
+@ExperimentalThumbnailsApi
+data class ThumbnailExtractParameters @JvmOverloads constructor(
+        /**
+         * The URI of the media from which to extract thumbnails.
+         */
+        val mediaUri: Uri,
+        /**
+         * The timestamp, in microseconds, for which to extract the frame. If the frame can't be extracted for a specified timestamp, the nearest frame may be
+         * returned: this frame selection behavior is defined in [mode].
+         */
+        val timestampUs: Long,
+        /**
+         * The renderer to use to process each of the thumbnails before they're returned.
+         *
+         * One instance of the renderer may be shared between requests, as long as the dimensions of the video frames (for media specified in [mediaUri]) and
+         * the [destSize] remain the same.
+         */
+        val renderer: ThumbnailRenderer,
+        /**
+         * A hint to the [ExtractionBehavior] about what approach to use for extracting video frames.
+         */
+        val mode: ExtractionMode = ExtractionMode.Fast,
+        /**
+         * Optional size of the generated thumbnail Bitmap. The video frame will be resized and scaled to fill this [destSize], preserving the video frame's
+         * original aspect ratio.
+         */
+        val destSize: Point? = null,
+        /**
+         * The optional priority of the request. Lower value indicates higher priority. Requests with equal priority will be handled in FIFO order.
+         */
+        val priority: Long = 0L
```

Comment:
Is there a range of these? Does this match thread executor priority api?

Reply:
The range is Long.MIN_VALUE..Long.MAX_VALUE.

This value is used to set the priority of this request relative to other requests you've submitted, so it's completely arbitrary. We could add more structure around this by e.g. making it an enum, though that seemed a little prescriptive (not sure how consumers would choose to set priority). 

We use a `ThreadPoolExecutor` with a `PriorityBlockingQueue` work queue. We call `execute()` with an instance of our `ComparableFutureTask` (which overrides `comparedTo` to use the priority value), and this task gets added to the `PriorityBlockingQueue`. So when the executor polls the next task, it would always be the one with the lowest `priority` value.

Reply:
Got it now, thanks!